File: bench/realworld/bench_json.js
// bench_json.js (qjs) - parse an 18 MB JSON array, ITERS times, summing value.
import * as std from "std";
var text = std.loadFile("bench_data.json");
var ITERS = 5;
var t0 = Date.now();
var total = 0;
for (var k = 0; k < ITERS; k++) {
    var arr = JSON.parse(text);
    for (var i = 0; i < arr.length; i++) total += arr[i].value;
}
var elapsed = Date.now() - t0;
print("result: " + total);
print("elapsed: " + elapsed + " ms");

File: bench/realworld/bench_json.nim
# bench_json.nim - parse an 18 MB JSON array, ITERS times, summing value fields.
import std/[json, times]

let text = readFile("bench_data.json")
let iters = 5
let t0 = epochTime()
var total: int64 = 0
for k in 0 ..< iters:
  let arr = parseJson(text)
  for rec in arr:
    total += rec["value"].getInt
let elapsed = int((epochTime() - t0) * 1000)
echo "result: ", total
echo "elapsed: ", elapsed, " ms"
